File: apps/frontend/services/chatService.ts
import api from "./api";
import { ChatMessage } from "@/types";

export const chatService = {
  getConversation: async (otherUserId: string): Promise<ChatMessage[]> => {
    const res = await api.get<ChatMessage[]>(`/api/chat/conversation/${otherUserId}`);
    return res.data;
  },

  getUnreadMessages: async (): Promise<ChatMessage[]> => {
    const res = await api.get<ChatMessage[]>("/api/chat/unread");
    return res.data;
  },

  markAsRead: async (senderId: string): Promise<void> => {
    await api.put(`/api/chat/read/${senderId}`);
  },

  isOnline: async (userId: string): Promise<boolean> => {
    const res = await api.get<boolean>(`/api/chat/status/${userId}`);
    return res.data;
  },
};

File: apps/chat-service/src/main/java/com/joborbit/chat/service/ChatService.java
package com.joborbit.chat.service;

import java.util.List;

import com.joborbit.chat.dto.ChatMessageDTO;

public interface ChatService {

    // Send message
    void sendMessage(ChatMessageDTO messageDTO);

    // Get conversation between two users
    List<ChatMessageDTO> getConversation(String user1, String user2);

    // Get unread messages
    List<ChatMessageDTO> getUnreadMessages(String userId);

    // Mark messages as read
    void markMessagesAsRead(String senderId, String receiverId);
}